mne/simulation/source.py

```diff
@@ -218,3 +220,126 @@ def simulate_stc(src, labels, stc_data, tmin, tstep, value_fun=None):
+    which is then combined with a BEM (can also be a sphere model dict) and
+    the sensor information in info to form a forward operator.
+
+    The source estimate object (with the forward operator) can be projected to
+    sensor-space using :func:`mne.simulation.evoked.simulate_evoked`.
+
+    Parameters
+    ----------
+    dipole : Dipole object
+        Instance of Dipole containing position, orientation and amplitude of
+        one or more dipoles. Several dipoles may occur simultaneously, but
+        when more than one time point is defined, the spacing between them
+        (i.e. the sampling rate) must be constant.
+    bem : str | dict
+        The BEM filename (str) or a loaded sphere model (dict).
+    info : dict
+        Sensor-info etc., e.g., from real evoked file.
+    trans : str | None
+        The head<->MRI transform filename. Must be provided unless BEM
+        is a sphere model.
+    min_dist : float
+        Minimum distance (in milimeters) from the dipole to the inner skull.
+    # model_meg : bool
+    #     Include MEG channels (with or without compensation) in model.
+    # model_eeg : bool
+    #     Include EEG channels in model.
+    n_jobs : int
+        Number of jobs to run in parallel (used in making forward solution).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    stc : instance of VolSourceEstimate
+        The dipoles converted to a discrete set of points and associated
+        time courses.
+    fwd : instance of Forward
+        The forward solution.
+
+    See Also
+    --------
+    mne.simulation.simulate_evoked
+
+    Notes
+    -----
+    .. versionadded:: 0.12.0
+    """
+    # Make copies to avoid mangling original dipole
+    tims = dipole.times.copy()
+    poss = dipole.pos.copy()
+    amps = dipole.amplitude.copy()
+    oris = dipole.ori.copy()
+
+    # Convert positions to discrete source space (allows duplicate rr & nn)
+    # NB information about dipole orientation enters here, then no more
+    sources = dict(rr=poss, nn=oris)
+    # Dipole objects must be in the head frame
+    sp = _make_discrete_source_space(sources, coord_frame='head')
+    src = SourceSpaces([sp])  # dict with working_dir, command_line not nec
+
+    # Forward operator created for channels in info (use pick_info to restrict)
+    fwd = make_forward_solution(info, trans, src, bem, fname=None,
+                                mindist=min_dist, ignore_ref=False,
+                                overwrite=False, n_jobs=n_jobs, verbose=None)
+    # Convert from free orientations to fixed (in-place)
+    convert_forward_solution(fwd, surf_ori=False, force_fixed=True,
+                             copy=False, verbose=None)
+
+    # Check for omissions due to mindist in only source space (0)
+    # NB arguably might be better simply to raise an error, rather than
+    # continuing with just a warning
```

Comment:
+1

the dipole fit should prevent dipoles too close to inner skull. it exposes a min_dist param.

I'd remove the min_dist parameter here and set it to 0 internally and raise an error is dipole is outside of inner skull.